### ISSUE TITLE
Load Testing in webpage nextjs. TypeError: request.endsWith is not a function.

### ISSUE THREAD

**Issue by MCode-Team** [oven-sh/bun#8723]
### What version of Bun is running?

bun -v1.0.26

### What platform is your computer?

uname -mprs x64

### What steps can reproduce the bug?

Load Testing in webpage nextjs require a lot. Show TypeError: request.endsWith is not a function. (In 'request.endsWith(".shared-runtime")', 'request.endsWith' is undefined) 
![1707211629470](https://github.com/oven-sh/bun/assets/11708283/b6104a01-3a3a-47db-8bbb-43ba991b4c6a)

It's the version bun -v1.0.26. but It's not he version bun -v1.0.22

**Comment by Electroid** [comment]
Would it be possible to share a minimal repro? Also if you could describe how many requests you're sending before you experience this error? (or if the error always occurs)

**Comment by felipemullen** [comment]
I just started seeing this issue in an existing site that was fine for the last few months up until this week. Not sure yet what is causing it, but if I find out I will update this ticket

**Comment by Jarred-Sumner** [comment]
It sounds like something is passing `undefined` to `require`. The question is what/why

**Comment by felipemullen** [comment]
Restarting my docker container fixed this (no rebuild), which is not a great answer or workaround. I am not sure how to reproduce it at this point, but for context, here is my dockerfile:

```dockerfile
FROM imbios/bun-node:18-slim AS base
ARG DEBIAN_FRONTEND=noninteractive

FROM base AS deps
WORKDIR /app

COPY package.json bun.lockb ./
RUN bun install

FROM base AS builder
WORKDIR /app
COPY --from=deps /app/node_modules ./node_modules
COPY . .

ENV NEXT_TELEMETRY_DISABLED 1
RUN bun run build

FROM base AS runner
WORKDIR /app

ENV NODE_ENV production
ENV NEXT_TELEMETRY_DISABLED 1

COPY --from=builder /app/public ./public

RUN mkdir .next
COPY --from=builder /app/.next/standalone ./
COPY --from=builder /app/.next/static ./.next/static

RUN chmod +x /usr/local/bin/docker-entrypoint.sh

ENV PORT 3000
EXPOSE 3000
ENV HOSTNAME "0.0.0.0"

CMD ["bun", "run", "server.js"]
```

If I encounter the issue again, I will try to enter the container to see if I find anything else.

<details>
  <summary>Open for full stack trace</summary>

<pre>
57 | }).bind(null, resolveFilename, hookPropertyMap);
58 | // This is a hack to make sure that if a user requires a Next.js module that wasn't bundled
59 | // that needs to point to the rendering runtime version, it will point to the correct one.
60 | // This can happen on `pages` when a user requires a dependency that uses next/image for example.
61 | mod.prototype.require = function(request) {
62 |     if (request.endsWith(".shared-runtime")) {
             ^
TypeError: request.endsWith is not a function. (In 'request.endsWith(".shared-runtime")', 'request.endsWith' is undefined)
      at /app/node_modules/next/dist/server/require-hook.js:62:9
      at getEdgeFunctionsPages (/app/node_modules/next/dist/server/next-server.js:926:26)
      at /app/node_modules/next/dist/server/next-server.js:885:17
      at renderErrorToResponseImpl (/app/node_modules/next/dist/server/next-server.js:873:37)
      at /app/node_modules/next/dist/server/base-server.js:2001:107
      at renderErrorToResponse (/app/node_modules/next/dist/server/base-server.js:2000:33)
      at /app/node_modules/next/dist/server/base-server.js:1988:36
      at /app/node_modules/next/dist/server/base-server.js:1987:33
      at /app/node_modules/next/dist/server/base-server.js:911:31
      at pipeImpl (/app/node_modules/next/dist/server/base-server.js:901:24)
      at /app/node_modules/next/dist/server/base-server.js:899:123
      at pipe (/app/node_modules/next/dist/server/base-server.js:898:20)
      at renderErrorImpl (/app/node_modules/next/dist/server/base-server.js:1983:27)
      at /app/node_modules/next/dist/server/base-server.js:1979:97
      at renderError (/app/node_modules/next/dist/server/base-server.js:1978:23)
      at renderError (/app/node_modules/next/dist/server/next-server.js:899:23)
      at /app/node_modules/next/dist/server/base-server.js:777:28
</pre>
</details>

**Comment by Fuchsoria** [comment]
Have same issue with nextjs 14.1, added guards patch but at the end it goes deeper into errors
patch: 
`    if (request && typeof request === "object" && request.endsWith(".shared-runtime")) {
        return originalRequire.call(this, `next/dist/server/future/route-modules/pages/vendored/contexts/${path.basename(request, ".shared-runtime")}`);
    }
    return originalRequire.call(this, request);`

![image](https://github.com/oven-sh/bun/assets/43413472/1ec13227-6aba-4cc0-ab20-c5e9eeeac6ae)

**Comment by GeitV** [comment]
We are getting this error as well, when running NextJS app with bun in docker. We are using custom API endpoint for K8S to do health check against pod and so this error happens as soon as the app deploys.

Our workaround was to run the app using nodejs in the meantime.

**Comment by Jarred-Sumner** [comment]
This is fixed in Bun v1.0.30 which releases in about an hour

In the meantime, you can upgrade with `bun upgrade --canary`
